edgetest_hub/plugin.py

```diff
@@ -203,11 +242,16 @@ def addoption(schema: Schema):
 @hookimpl
 def post_run_hook(testers: List, conf: Dict):
     """Invoke hub after the testing is complete."""
-    if (
-        testers[-1].status and GIT_TOKEN_ENVNAME in os.environ
-    ):  # skip is token not available
-        if conf.get("hub"):
-            configure_branch(conf)
-            push_branch(conf)
+    if GIT_TOKEN_ENVNAME in os.environ:
+        if testers[-1].status is True:
+            if conf.get("hub"):
+                configure_branch(conf)
+                push_branch(conf)
+        else:  # testers[-1].status is False
+            if conf["hub"]["open_issue_on_fail"] is True:
```

Comment:
Will this produce a `KeyError` for the case where `edgetest-hub` is not configured, but `GIT_TOKEN_ENVNAME` exists in the environment? I have not worked with plugin managers much before so this may not be a practical edge case that would arise, since if this code is being executed then by-default `conf['hub']` may exist. If so, I believe it could be addressed by adding an extra layer of logic:

```python
if GIT_TOKEN_ENVNAME in os.environ:
  if testers[-1].status is True:
    ...
  else: # testers[-1].status is False
    if conf.get("hub"):
      if conf["hub"]["open_issue_on_fail"] is True:
        ...
      else:
        LOG.info("Skipping Creating an Issue.")
    else:
      LOG.info("Hub plugin configuration not found. Skipping Hub plugin")
else:
  ...
```

Reply:
Ahh good point. I think regardless we can just add the check in to be explicit. I'll add in the change.